program-runtime/src/invoke_context.rs

```diff
@@ -209,6 +214,17 @@ impl<'a> InvokeContext<'a> {
             .or_else(|| self.programs_loaded_for_tx_batch.find(pubkey))
     }
 
+    pub fn get_environments_for_slot(
+        &self,
+        slot: Slot,
+    ) -> Result<&ProgramRuntimeEnvironments, InstructionError> {
+        let epoch_schedule = self.sysvar_cache.get_epoch_schedule()?;
```

Comment:
if we're going to hit the sysvar cache anyway, why not just pull the epoch directly out of `Clock`

Reply:
We need epoch corresponding to the `effective_slot` for the program being deployed. So it doesn't correspond to the current `clock`.

Reply:
ah. can you rename the param to `effective_slot` in that case?

Reply:
Yes, that's a good call. Will do.